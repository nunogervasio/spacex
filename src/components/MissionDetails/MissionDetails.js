import React from "react";
import PropTypes from "prop-types";
import styles from "./MissionDetails.module.css";

const MissionDetails = ({
  name,
  number,
  date,
  site,
  rocketName,
  rocketType,
  details
}) => {
  return (
    <p>
      <span className={styles.flight_name}>{name}</span>
      <br />
      <strong>Flight Number:</strong> {number}
      <br />
      <strong>Date:</strong> {date}
      <br />
      <strong>Location:</strong> {site}
      <br />
      <strong>Rocket Name:</strong> {rocketName}
      <br />
      <strong>Rocket Type:</strong> {rocketType}
      <br />
      <strong>Mission Details:</strong> {details}
    </p>
  );
};

export default MissionDetails;

MissionDetails.propTypes = {
  name: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  site: PropTypes.string.isRequired,
  rocketName: PropTypes.string.isRequired,
  rocketType: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired
};
