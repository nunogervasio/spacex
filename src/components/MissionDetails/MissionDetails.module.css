.flight_name {
  font-size: 1.5em;
  vertical-align: baseline;
  line-height: 50%;
}
