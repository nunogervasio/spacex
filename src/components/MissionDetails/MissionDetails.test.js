import React from "react";
import { shallow } from "enzyme";
import MissionDetails from "./MissionDetails";

describe("Mission Details", () => {
  const props = {
    name: "",
    number: "",
    date: "",
    site: "",
    rocketName: "",
    rocketType: "",
    details: ""
  };
  const missionDetails = shallow(<MissionDetails {...props} />);
  it("renders properly", () => {
    expect(missionDetails).toMatchSnapshot();
  });
});
