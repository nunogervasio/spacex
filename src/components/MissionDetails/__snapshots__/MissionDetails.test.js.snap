// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Details renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MissionDetails
    date=""
    details=""
    name=""
    number=""
    rocketName=""
    rocketType=""
    site=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="flight_name"
        >
          
        </span>,
        <br />,
        <strong>
          Flight Number:
        </strong>,
        " ",
        "",
        <br />,
        <strong>
          Date:
        </strong>,
        " ",
        "",
        <br />,
        <strong>
          Location:
        </strong>,
        " ",
        "",
        <br />,
        <strong>
          Rocket Name:
        </strong>,
        " ",
        "",
        <br />,
        <strong>
          Rocket Type:
        </strong>,
        " ",
        "",
        <br />,
        <strong>
          Mission Details:
        </strong>,
        " ",
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "flight_name",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Flight Number:",
        },
        "ref": null,
        "rendered": "Flight Number:",
        "type": "strong",
      },
      " ",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Date:",
        },
        "ref": null,
        "rendered": "Date:",
        "type": "strong",
      },
      " ",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Location:",
        },
        "ref": null,
        "rendered": "Location:",
        "type": "strong",
      },
      " ",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rocket Name:",
        },
        "ref": null,
        "rendered": "Rocket Name:",
        "type": "strong",
      },
      " ",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rocket Type:",
        },
        "ref": null,
        "rendered": "Rocket Type:",
        "type": "strong",
      },
      " ",
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mission Details:",
        },
        "ref": null,
        "rendered": "Mission Details:",
        "type": "strong",
      },
      " ",
      "",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="flight_name"
          >
            
          </span>,
          <br />,
          <strong>
            Flight Number:
          </strong>,
          " ",
          "",
          <br />,
          <strong>
            Date:
          </strong>,
          " ",
          "",
          <br />,
          <strong>
            Location:
          </strong>,
          " ",
          "",
          <br />,
          <strong>
            Rocket Name:
          </strong>,
          " ",
          "",
          <br />,
          <strong>
            Rocket Type:
          </strong>,
          " ",
          "",
          <br />,
          <strong>
            Mission Details:
          </strong>,
          " ",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "flight_name",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Flight Number:",
          },
          "ref": null,
          "rendered": "Flight Number:",
          "type": "strong",
        },
        " ",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Date:",
          },
          "ref": null,
          "rendered": "Date:",
          "type": "strong",
        },
        " ",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Location:",
          },
          "ref": null,
          "rendered": "Location:",
          "type": "strong",
        },
        " ",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rocket Name:",
          },
          "ref": null,
          "rendered": "Rocket Name:",
          "type": "strong",
        },
        " ",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rocket Type:",
          },
          "ref": null,
          "rendered": "Rocket Type:",
          "type": "strong",
        },
        " ",
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mission Details:",
          },
          "ref": null,
          "rendered": "Mission Details:",
          "type": "strong",
        },
        " ",
        "",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
